### PR TITLE
Report the values of `_Py_HashSecret` and `random.seed()` in test failures

### PR DESCRIPTION
**Pull request by elprans**
This should hopefully help with the diagnostics of extreme Heisenbugs.
Unfortunately, it is not possible to directly specify the exact value of
`_Py_HashSecret`, so a minor CPython patch would be necessary to make
this data useful.

### CODE REVIEW

File: edb/tools/test/cpython_hacks.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+class _Py_HashSecret_t(ctypes.Union):
+    _fields_ = [
+        ('uc', ctypes.c_ubyte * 24),
+    ]
+
+
+def get_py_hash_secret() -> bytes:
+    hashsecret = _Py_HashSecret_t.in_dll(ctypes.pythonapi, '_Py_HashSecret')
+    return bytes(hashsecret.uc)
```
Comment (1st1) [inline edb/tools/test/cpython_hacks.py:34]: Do we need to care about 32 bit builds? (I'm not if _Py_HashSecret_t is defined differently for those)

File: edb/tools/test/cpython_hacks.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#
```
Comment (1st1) [inline edb/tools/test/cpython_hacks.py:1]: Can we please give a descriptive name to this module -- `secret_hash.py` or something.

**Review by 1st1 — APPROVED** [review]

File: edb/tools/test/cpython_hacks.py (hunk 2 above)
Comment (msullivan) [reply edb/tools/test/cpython_hacks.py:1]: maybe in a `cpython_hacks` directory <_<

Comment (elprans) [reply edb/tools/test/cpython_hacks.py:1]: I think cpython_hacks is descriptive here as we are reaching into the interpreter internals specific to cpython and it's likely this wouldn't be the last such hack

Comment (1st1) [reply edb/tools/test/cpython_hacks.py:1]: I disagree. The hack part is an implementation detail, I couldn't care less how a module works.

File: edb/tools/test/cpython_hacks.py (hunk 1 above)
Comment (elprans) [reply edb/tools/test/cpython_hacks.py:34]: It is always 24 bytes long: https://github.com/python/cpython/blob/b2f68b190035540872072ac1d2349e7745e85596/Include/pyhash.h#L56-L57

**Comment by elprans** [comment]
Here's a CPython patch to go along with this: https://github.com/elprans/cpython/commit/0cece3de55b26d247ebaaf7c811f93c2f8d0a268

File: edb/tools/test/cpython_hacks.py (hunk 2 above)
Comment (elprans) [reply edb/tools/test/cpython_hacks.py]: Renamed to `cpython_state`. I'd like to avoid naming the whole module after the function.

Comment (1st1) [reply edb/tools/test/cpython_hacks.py]: Sure, that's a good name!

**Pull request merged**